lib/gemstash/s3.rb

```diff
@@ -0,0 +1,345 @@
+require "gemstash"
+require "digest"
+require "fileutils"
+require "pathname"
+require "yaml"
+require "aws-sdk"
+
+module Gemstash
```

Comment:
There's a lot going on in this class. Could we move out some of the different parts into separate files?